component/otelcol/config_grpc.go

```diff
@@ -0,0 +1,122 @@
+package otelcol
+
+import (
+	"time"
+
+	otelconfiggrpc "go.opentelemetry.io/collector/config/configgrpc"
+	"go.opentelemetry.io/collector/config/confignet"
+)
+
+// GRPCServerArguments holds shared gRPC settings for components which launch
+// gRPC servers.
+type GRPCServerArguments struct {
+	Endpoint  string `river:"endpoint,attr,optional"`
+	Transport string `river:"transport,attr,optional"`
+
+	TLS *TLSServerArguments `river:"tls,block,optional"`
+
+	MaxRecvMsgSizeMiB    uint64 `river:"max_recv_msg_size_mib,attr,optional"`
+	MaxConcurrentStreams uint32 `river:"max_concurrent_streams,attr,optional"`
+	ReadBufferSize       int    `river:"read_buffer_size,attr,optional"`
+	WriteBufferSize      int    `river:"write_buffer_size,attr,optional"`
```

Comment:
After a discussion offline, it'd be nice to have these be River strings which describe the unit (`"500B"`).

Ditto for other config types.